projects/step-frontend/src/lib/modules/timeseries/performance-view/chart-generators/chart-generators.ts

```diff
@@ -0,0 +1,99 @@
+import { FindBucketsRequest } from '../../find-buckets-request';
+import { TimeSeriesUtils } from '../../time-series-utils';
+import { UPlotUtils } from '../../uplot/uPlot.utils';
+import { TimeSeriesChartResponse } from '../../time-series-chart-response';
+import { TSChartSettings } from '../../chart/model/ts-chart-settings';
+import { TsChartType } from '../ts-chart-type';
+
+declare const uPlot: any;
+
+export class ChartGenerators {
+  private static readonly CHART_LEGEND_SIZE = 65;
+
+  private static readonly barsFunction = uPlot.paths.bars; // this is a function from uplot which allows to draw bars instead of straight lines
+  private static readonly stepped = uPlot.paths.stepped; // this is a function from uplot wich allows to draw 'stepped' or 'stairs like' lines
+
+  static generateChart(
+    type: TsChartType,
+    request: FindBucketsRequest,
+    response: TimeSeriesChartResponse
+  ): TSChartSettings {
+    switch (type) {
+      case TsChartType.OVERVIEW:
```

Comment:
At some point we could have registry mechanism where custom charts register themselves. This would allow plugins to register custom charts and custom dashboards to reference them